internal/providers/github/properties/release.go

```diff
@@ -0,0 +1,154 @@
+	ReleasePropertyRepo = "github/repo"
+	// ReleasePropertyTag represents the github release tag name.
+	ReleasePropertyTag = "github/tag"
+	// ReleasePropertyBranch represents the github release branch
+	ReleasePropertyBranch = "github/branch"
+)
+
+// ReleaseFetcher is a property fetcher for releases
+type ReleaseFetcher struct {
+	propertyFetcherBase
+}
+
+// NewReleaseFetcher creates a new ReleaseFetcher
+func NewReleaseFetcher() *ReleaseFetcher {
+	return &ReleaseFetcher{
+		propertyFetcherBase: propertyFetcherBase{
+			propertyOrigins: []propertyOrigin{
+				{
+					keys: []string{
+						// general entity
+						properties.PropertyName,
+						properties.PropertyUpstreamID,
+						// general release
+						ReleasePropertyTag,
+						ReleasePropertyBranch,
+					},
+					wrapper: getReleaseWrapper,
+				},
+			},
+			operationalProperties: []string{},
+		},
+	}
+}
+
+// GetName returns the name of the release
+func (_ *ReleaseFetcher) GetName(props *properties.Properties) (string, error) {
+	owner := props.GetProperty(ReleasePropertyOwner).GetString()
+	repo, err := props.GetProperty(ReleasePropertyRepo).AsString()
+	if err != nil {
+		return "", fmt.Errorf("failed to get repo name: %w", err)
+	}
+
+	tag, err := props.GetProperty(ReleasePropertyTag).AsString()
+	if err != nil {
+		return "", fmt.Errorf("failed to get tag name: %w", err)
+	}
+
+	return getReleaseNameFromParams(owner, repo, tag), nil
+}
+
+func getReleaseNameFromParams(owner, repo, tag string) string {
+	if owner == "" {
+		return fmt.Sprintf("%s/%s", repo, tag)
+	}
+
+	return fmt.Sprintf("%s/%s/%s", owner, repo, tag)
+}
+
+func getReleaseWrapper(
+	ctx context.Context, ghCli *go_github.Client, _ bool, getByProps *properties.Properties,
+) (map[string]any, error) {
+	// TODO: Should I be parsing this as string or int64?
+	// if string, then I should convert it to int64
+	upstreamID, err := getByProps.GetProperty(properties.PropertyUpstreamID).AsInt64()
+	if err != nil {
+		return nil, fmt.Errorf("upstream ID not found or invalid: %w", err)
+	}
+
+	owner, err := getByProps.GetProperty(ReleasePropertyOwner).AsString()
+	if err != nil {
+		return nil, fmt.Errorf("owner not found or invalid: %w", err)
+	}
+
+	repo, err := getByProps.GetProperty(ReleasePropertyRepo).AsString()
+	if err != nil {
+		return nil, fmt.Errorf("repo not found or invalid: %w", err)
+	}
+
+	var fetchErr error
+	var release *go_github.RepositoryRelease
+	var result *go_github.Response
+	release, result, fetchErr = ghCli.Repositories.GetRelease(ctx, owner, repo,
+		upstreamID)
+	if fetchErr != nil {
+		if result != nil && result.StatusCode == http.StatusNotFound {
+			return nil, v1.ErrEntityNotFound
+		}
+		return nil, fmt.Errorf("failed to fetch release: %w", fetchErr)
+	}
+
+	return map[string]any{
+		properties.PropertyUpstreamID: properties.NumericalValueToUpstreamID(release.GetID()),
+		properties.PropertyName:       getReleaseNameFromParams(owner, repo, release.GetTagName()),
```

Comment:
It looks like GitHub's `name` here is non-unique, but may still be worth exposing for e.g. filtering with CEL.  (Apply / omit evalution only to releases with certain names)